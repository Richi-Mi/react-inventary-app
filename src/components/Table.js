import React, { Fragment, useState } from 'react';
import FormAdd from './FormAdd';
import RowCells from './RowCells';

const Table = ({ defaultCells = []}) => {
    const [cells, setCells] = useState( defaultCells );
    // Se hara un map de cells y se agregara un <tr> por cada iteraci√≥n
    return (
        <Fragment>
            <header className='header'>
                <h1 className='title_principal'> Inventary </h1>
                <img src='./logo_inventary.jpg' alt='inventary' />
            </header>

            <FormAdd setCells={ setCells } cells={ cells } />

            <table className='table' id='table_inventary'>
                <tbody className='tconten'>
                    <tr className='row row_title'>
                        <td> ID </td>
                        <td> Nombre </td>
                        <td> Cantidad </td>
                        <td> Precio </td>
                        <td> Dinero Total </td>
                    </tr>
                    {
                        cells.map(({ name, units, price }, i) => {
                            let unidades = parseInt( units );
                            let precio = parseInt( price );
                            
                            return <RowCells name={ name } 
                                            units={ unidades } 
                                            price={ precio } 
                                            key={`row_cells_${i}`} 
                                            id={i} />
                            
                        })
                    }
                </tbody>
            </table>
        </Fragment>
    )
}
export default Table