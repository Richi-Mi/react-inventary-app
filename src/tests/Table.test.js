import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import Table from '../components/Table';

describe(' * Pruebas en el componente <Table /> ', () => {
    // Usar beforeEach()
    test(' - Debe Mostrar el Componente Table Correctamente', () => {
        const wrapper = shallow( <Table /> ); 
        expect( wrapper ).toMatchSnapshot();
    });
    test(' - Debe Mostrar las Celdas Correctamente ', () => {
        const cells = [
            {
                name: 'Galletas', 
                units: 4, 
                price: 200
            },
            {
                name: 'Pan Danes', 
                units: 2, 
                price: 50
            },
            {
                name: 'Pastel', 
                units: 1, 
                price: 500
            }
        ]
        const wrapper = shallow( <Table defaultCells={ cells }/> ); 
        const rowsCells = wrapper.find('RowCells');

        expect( rowsCells.length ).toBe( cells.length );

    })
})