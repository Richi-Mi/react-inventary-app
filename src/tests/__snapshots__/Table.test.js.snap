// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` * Pruebas en el componente <Table />   - Debe Mostrar el Componente Table Correctamente 1`] = `
<Fragment>
  <header
    className="header"
  >
    <h1
      className="title_principal"
    >
       Inventary 
    </h1>
    <img
      alt="inventary"
      src="./logo_inventary.jpg"
    />
  </header>
  <FormAdd
    cells={Array []}
    setCells={[Function]}
  />
  <table
    className="table"
    id="table_inventary"
  >
    <tbody
      className="tconten"
    >
      <tr
        className="row row_title"
      >
        <td>
           ID 
        </td>
        <td>
           Nombre 
        </td>
        <td>
           Cantidad 
        </td>
        <td>
           Precio 
        </td>
        <td>
           Dinero Total 
        </td>
      </tr>
    </tbody>
  </table>
</Fragment>
`;
